src/OpenTelemetry/Trace/Configuration/TracerFactory.cs

```diff
@@ -0,0 +1,265 @@
+﻿// <copyright file="TracerFactory.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using OpenTelemetry.Context.Propagation;
+    using OpenTelemetry.Resources;
+    using OpenTelemetry.Trace.Export;
+    using OpenTelemetry.Trace.Sampler;
+
+    public class TracerBuilder : IDisposable
```

Comment:
I don't think this should implement IDisposable.

Builder should be created to builder the factory, and then thrown away. The state is on the factory, and it is what should be help onto and disposed.